rai_test_utils/rai_test_utils/utilities/utils.py

```diff
@@ -62,3 +73,113 @@ def retrieve_dataset(dataset, **kwargs):
         return pd.read_csv(filepath, **kwargs)
     else:
         raise Exception('Unrecognized file extension: ' + extension)
+
+
+def _get_retry_session(url):
+    domain = urlparse(url.lower()).netloc
+    if domain in _requests_sessions:
+        return _requests_sessions[domain]
+
+    session = requests.Session()
+    retries = Retry(
+        total=5, backoff_factor=1, status_forcelist=[500, 502, 503, 504]
+    )
+    session.mount("http://", HTTPAdapter(max_retries=retries))
+    session.mount("https://", HTTPAdapter(max_retries=retries))
+    _requests_sessions[domain] = session
+
+    return session
+
+
+def get_image_from_path(image_path, image_mode):
+    """Get image from path.
+
+    :param image_path: The path to the image.
+    :type image_path: str
+    :param image_mode: The mode to open the image in.
+        See pillow documentation for all modes:
+        https://pillow.readthedocs.io/en/stable/handbook/concepts.html
+    :type image_mode: str
+    :return: The image as a numpy array.
+    :rtype: numpy.ndarray
+    """
+    image_open_pointer = image_path
+    if image_path.startswith("http://") or image_path.startswith("https://"):
+        response = _get_retry_session(image_path).get(image_path)
+        image_open_pointer = BytesIO(response.content)
+    with Image.open(image_open_pointer) as im:
+        if image_mode is not None:
+            im = im.convert(image_mode)
+        image_array = np.asarray(im)
+    return image_array
+
+
+def convert_images(dataset, image_mode):
+    """Converts the images to the format required by the model.
+
+    If the images are base64 encoded, they are decoded and converted to
+    numpy arrays. If the images are already numpy arrays, they are
+    returned as is.
+
+    :param dataset: The dataset to convert.
+    :type dataset: numpy.ndarray
+    :param image_mode: The mode to open the image in.
+        See pillow documentation for all modes:
+        https://pillow.readthedocs.io/en/stable/handbook/concepts.html
+    :type image_mode: str
+    :return: The converted dataset.
+    :rtype: numpy.ndarray
+    """
+    if len(dataset) > 0 and isinstance(dataset[0], str):
+        try:
+            dataset = np.array([get_image_from_path(
+                x, image_mode) for x in dataset])
+        except ValueError:
+            # if images of different sizes, try to convert one by one
+            jagged = np.empty(len(dataset), dtype=object)
+            for i, x in enumerate(dataset):
+                jagged[i] = get_image_from_path(x, image_mode)
+            dataset = jagged
+    return dataset
+
+
+def get_images(dataset, image_mode, transformations=None):
```

Comment:
Could this method and its dependencies live in dataset/vision? Looks like these methods are specific to vision so not sure if these should be here.